src/dashboard-refactor/lists-sidebar/components/lists-sidebar-item/index.tsx

```diff
@@ -116,47 +108,39 @@ const NewItemsCountInnerDiv = styled.div`
     line-height: 14px;
 `
 
-// this type is differentiated from the type which governs the object passed down the tree to its parent:
-// the click handlers in this type have received their parameters from the parent and so receive none
-export interface ListsSidebarItemComponentProps {
-    className?: string
-    listName: string
-    isEditing: boolean
-    hoverState: HoverState
-    selectedState: SelectedState
-    dropReceivingState: DropReceivingState
-    newItemsCountState: NewItemsCountState
-    moreActionButtonState: MoreActionButtonState
-}
-
-interface HoverState {
-    onHoverEnter(): void
-    onHoverLeave(): void
-    isHovered: boolean
-}
-
-interface SelectedState {
-    onSelection(): void
-    isSelected: boolean
-}
-
-export interface DropReceivingState {
-    onDragOver(): void
-    onDragLeave(): void
-    onDrop(): void
-    triggerSuccessfulDropAnimation: boolean
-    isDraggedOver: boolean
-    canReceiveDroppedItems: boolean
-}
-
-interface MoreActionButtonState {
-    onMoreActionClick(): void
-    displayMoreActionButton
-}
-
 export default class ListsSidebarItem extends PureComponent<
     ListsSidebarItemComponentProps
 > {
+    private renderListName = () => {
+        const {
+            listName,
+            hoverState: { isHovered },
+            selectedState: { isSelected },
+        } = this.props
+        if (typeof listName === 'string')
+            return (
+                <ListTitle isHovered={isHovered} isSelected={isSelected}>
+                    {listName}
+                </ListTitle>
+            )
+        let spanArray: JSX.Element[] = []
+        listName.forEach(({ text, match }) =>
```

Comment:
This is a good candidate for Array.map instead of Array.forEach

e.g.:
```ts
const spanArray = listName.map(({ text, match }) => match ? (...) : (...))
```